File: Back-end/src/main/java/by/bsuir/em/service/TestService.java
package by.bsuir.em.service;

import by.bsuir.em.dto.TestDto;

import java.util.List;

public interface TestService {
    List<TestDto> getTests();
    TestDto getTestById(Long id);
    List<TestDto> getTestsByNameFragment(String nameFragment);
    TestDto addTest(TestDto testDto);
    TestDto updateTest(TestDto testDto);
    void deleteTest(Long id);
}

File: Front-end/app/common/service/test.service.js
export default function testService($http) {
    var service = this;

    service.getTests = function () {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/tests"
        }).then(function (response) {
            return response.data;
        });
    };

    service.getTestById = function (id) {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/tests/" + id
        }).then(function (response) {
            return response.data;
        });
    };

    service.getTestsByNameFragment = function (nameFragment) {
        return $http({
            method: "GET",
            url: EM_REST_SERVER_URI + "/tests/?name=" + nameFragment
        }).then(function (response) {
            return response.data;
        });
    };

    service.addTest = function (test) {
        return $http({
            method: "POST",
            url: EM_REST_SERVER_URI + "/tests",
            data: test
        }).then(function (response) {
            return response.data;
        });
    };
}